config/database.yml

```diff
@@ -20,10 +20,14 @@ default: &default
   # For details on connection pooling, see Rails configuration guide
   # https://guides.rubyonrails.org/configuring.html#database-pooling
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  host: <%= ENV.fetch("DB_HOST", "localhost") %>
+  username: <%= ENV.fetch("DB_USERNAME", "postgres") %>
+  port: <%= ENV.fetch("DB_PORT", "5432") %>
 
 development:
   <<: *default
-  database: learning_journal_development
+  database: <%= ENV.fetch("DB_NAME", "learning_journal_development") %>
+  password: <%= ENV.fetch("DB_PASSWORD", "notinproduction") %>
```

Comment:
This line is very important and exists only dev and test environments. The docker image needs a password by default, which is set when at the docker image level (see the `docker-compose.yml` file, below). I am instructing rails that the default value is `notinproduction` here. This is safe to disclose here as these are not used in a production/online environment.

It is absolutely crucial to understand that I would not put a password like this directly into the file unless I was sure it is local only.

Reply:
But isn't it still being pushed to GitHub, and therefore made public? This file is not excluded from being pushed to the cloud by .gitignore